src/components/Form/components/CVCInput.tsx

```diff
@@ -1,15 +1,14 @@
-import { ChangeEvent, Dispatch } from "react";
+import { ChangeEvent } from "react";
 import { EFormaActions, EFormData } from "../../../enums/Form";
 import { formatCVC } from "../../../helpers/helpers";
-import { FormActions } from "../../../reducers/form-reducer";
+import { useForm } from "../../../hooks/useForm";
 
-type CVCInputProps = {
-  state: string;
-  dispatch: Dispatch<FormActions>;
-  error?: string;
-};
+const CVCInput = () => {
+
+  const { state, dispatch, errors } = useForm()
+  const { cvc } = state
+  const cvcError = errors[EFormData.CVC];
 
-const CVCInput = ({ state, dispatch, error }: CVCInputProps) => {
   const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component